hw3/ThronesAPI/src/HousesChart.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
+import { Doughnut } from "react-chartjs-2";
+import { Container } from "react-bootstrap";
+import NavbarComponent from "./NavbarComponent";
+
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+const backgroundColors = [
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+];
+
+const borderColors = [
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+];
+
+const url = "https://thronesapi.com/api/v2/Characters";
+
+const HousesChart = () => {
+  const [houseCounts, setHouseCounts] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(url);
+        const apiData = response.data;
```

Comment:
Calls the same API and fetches the same data as that of CharacterPage.jsx

Fetching the data (calling the exact same API) twice, in both the CharacterPage and HousesChart components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent.

So why not let the data flow from parent to child?

It would make a lot more sense to call the API once in the App component, then pass the data as props to the CharacterPage and HousesChart components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.